### PR TITLE
Improve CassandraClientPoolingContainer metrics and settings

### PR DESCRIPTION
**Pull request by diogoholanda**
**Goals (and why)**:
I am investigating some potential resource not being closed by the Cassandra integration.
I am interested in how many clients get created/evicted and when does that happen.
Those questions are hard to answer with the current metrics.

**Implementation Description (bullets)**:
Produce the desired metrics.

### CODE REVIEW

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolingContainer.java [hunk 1]
```diff
@@ -257,7 +257,6 @@ public String toString() {
 
         // immediately throw when we try and borrow from a full pool; dealt with at higher level
         poolConfig.setBlockWhenExhausted(false);
-        poolConfig.setMaxWaitMillis(config.socketTimeoutMillis());
```
Comment (diogoholanda) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolingContainer.java]: `MaxWaitMillis` is ignored when `BlockWhenExhausted` is false.
This configuration means how long to block if the pool is exhausted.
It is also weird that this was getting set to the socket timeout millis. This seemed like an oversight

Comment (diogoholanda) [reply atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolingContainer.java]: Note that this was originally added in a PR when `BlockWhenExhausted` was true: https://github.com/palantir/atlasdb/commit/99de2f76d2ea9b47ba16b3eb0c3ffc52970d3278#diff-b88d7ebe9dabc333cf7e4dff942e1830R149

Comment (jeremyk-91) [reply atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolingContainer.java]: Yep, confirming that this is correct - it is indeed ignored (from `GenericObjectPool::borrowObject`)

**Comment by jeremyk-91** [comment]
will add no-changelog tag

**Comment by diogoholanda** [comment]
Thanks! Can you merge this @jeremyk-91 since I don't have the permissions needed?
I would also appreciate a release if that's possible, so that I can continue my investigation

**Pull request merged**